app/code/Magento/Checkout/Test/Unit/Model/Cart/CollectQuoteTest.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Checkout\Test\Unit\Model\Cart;
+
+use Magento\Customer\Api\AddressRepositoryInterface;
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Customer\Api\Data\AddressInterface;
+use Magento\Customer\Api\Data\CustomerInterface;
+use Magento\Customer\Api\Data\RegionInterface;
+use Magento\Checkout\Model\Cart\CollectQuote;
+use Magento\Customer\Model\Session as CustomerSession;
+use Magento\Quote\Api\CartRepositoryInterface;
+use Magento\Quote\Api\Data\EstimateAddressInterface;
+use Magento\Quote\Api\Data\EstimateAddressInterfaceFactory;
+use Magento\Quote\Api\ShippingMethodManagementInterface;
+use Magento\Quote\Model\Quote;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class CollectQuoteTest
+ */
+class CollectQuoteTest extends TestCase
+{
+    /**
+     * @var CollectQuote
+     */
+    private $model;
+
+    /**
+     * @var CustomerSession|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $customerSessionMock;
+
+    /**
+     * @var CustomerRepositoryInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $customerRepositoryMock;
+
+    /**
+     * @var AddressRepositoryInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $addressRepositoryMock;
+
+    /**
+     * @var EstimateAddressInterfaceFactory|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $estimateAddressFactoryMock;
+
+    /**
+     * @var EstimateAddressInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $estimateAddressMock;
+
+    /**
+     * @var ShippingMethodManagementInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $shippingMethodManagerMock;
+
+    /**
+     * @var CartRepositoryInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $quoteRepositoryMock;
+
+    /**
+     * @var Quote|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $quoteMock;
+
+    /**
+     * @var CustomerInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $customerMock;
+
+    /**
+     * @var AddressInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $addressMock;
+
+    /**
+     * Set up
+     */
+    protected function setUp()
+    {
+        $this->customerSessionMock = $this->createMock(CustomerSession::class);
+        $this->customerRepositoryMock = $this->getMockForAbstractClass(
+            CustomerRepositoryInterface::class,
+            [],
+            '',
+            false,
+            true,
+            true,
+            ['getById']
+        );
+        $this->addressRepositoryMock = $this->createMock(AddressRepositoryInterface::class);
+        $this->estimateAddressMock = $this->createMock(EstimateAddressInterface::class);
+        $this->estimateAddressFactoryMock = $this->getMockBuilder(EstimateAddressInterfaceFactory::class)
```

Comment:
You can simplify this part of the code by using `createPartialMock` instead of the chain with `disableOriginalConstructor` and other methods.